applications/services/gui/modules/number_input.c

```diff
@@ -0,0 +1,493 @@
+#include "number_input.h"
+
+#include <gui/elements.h>
+#include <furi.h>
+#include <assets_icons.h>
+
+/** NumberInput type */
+struct NumberInput {
+    View* view;
+};
+
+typedef struct {
+    const char text;
+    const uint8_t x;
+    const uint8_t y;
+} NumberInputKey;
+
+typedef struct {
+    const char* header;
+    FuriString* text_buffer;
+    bool clear_default_text;
+    int32_t max_value;
+    int32_t min_value;
+    bool useSign;
+    char sign;
+    NumberInputCallback callback;
+    void* callback_context;
+    int8_t selected_row;
```

Comment:
Can it be negative? if not then size_t. In general there is no sense in using uintX_t types where specialized one exists

Reply:
Your right, as it cannot be negative, there's no need for a sign. 

I'm having trouble figuring out why **size_t** is preferred over **uint8_t**. Can you explain this? From what I can gather it's just uint8_t is alway 8-Bit while the Bit-size of size_t are platform dependent.

Reply:
It's all about machine word size and architecture. If you have 32bit Arm MCU then it will operate with 32bit wide registers. So basically you place your 8bit value to 32bit register do things and then transfer it back. That's not going to be faster then doing same thing with 32bit wide variable.

Also 8 bit is kinda small by itself, make no sense to put such constrains on software that meant to be portable and future proof.

Keep